From 69ad29bdad39878c70eeb07ec5e2f66e25908078 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Tue, 20 Oct 2026 15:10:36 +0200
Subject: [PATCH] avcodec/jacosubdec: do not read the event beyond the packet

Fixes: a3ewdvDE2QSe
Fixes: out of array read
Out of array read Replicated through API with ASAN
Found during triage of the review of https://code.ffmpeg.org/FFmpeg/FFmpeg/pulls/24659

(cherry picked from commit 69ad29bdad39878c70eeb07ec5e2f66e25908078)
---
 libavcodec/jacosubdec.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/libavcodec/jacosubdec.c b/libavcodec/jacosubdec.c
index fd90dcc4b69..c9fdfc2627c 100644
--- a/libavcodec/jacosubdec.c
+++ b/libavcodec/jacosubdec.c
@@ -27,6 +27,7 @@
 #include <time.h>
 #include "ass.h"
 #include "codec_internal.h"
+#include "decode.h"
 #include "jacosub.h"
 #include "libavutil/avstring.h"
 #include "libavutil/bprint.h"
@@ -164,11 +165,13 @@ static int jacosub_decode_frame(AVCodecContext *avctx, AVSubtitle *sub,
                                 int *got_sub_ptr, const AVPacket *avpkt)
 {
     int ret;
-    const char *ptr = avpkt->data;
+    const char *ptr = ff_decode_subtitle_text(avctx, avpkt);
     FFASSDecoderContext *s = avctx->priv_data;
 
     if (avpkt->size <= 0)
         goto end;
+    if (!ptr)
+        return AVERROR(ENOMEM);
 
     if (*ptr) {
         AVBPrint buffer;
-- 
2.53.0

